### DISCUSSION TITLE
Reading / watching apollo cache object for changes

### DISCUSSION THREAD

**Discussion by axos88** [the-guild-org/apollo-angular#1551] [General]
Hi!

I have a query that retrieves a user profile and their photos. The photos may or may not be processed at that time, but the server emits a notification when that happens and the client subscribes to those events to know when a photo has been processed.

The userProfile result has photo objects nested deeply, but the normalization process takes care of that, and puts the photo:<id> objects in the cache. The subscription handler can update them correctly.

In another (decoupled) part of the code (image component) I would like to either poll, or wait for one or multiple photos to be processed and to be presentable to the user. The input to the component is a photo id

How can I write a query/watchquery that would never reach the server, but would return the photo object in the cache?
I can't use `.readQuery`, because there is no "real" query for fetching photos directly. 

Basically I would need a `cache.watch("photo:<ID>")` that returns an Observable<Photo>

**Comment by axos88** [comment]
```
query userProfile {
  userProfile(ids: ['aaa','bbb']) {
    id
    (...)
    photos: {
      id
      signedUrl
      processed_at
    }
  }
}
```

Cache:

```
ROOT_QUERY
Photo:5f7a2ec6-f078-4aea-9f3e-d9a8f0cfe64e
Photo:7660a7dd-fc4f-4e4b-9659-9d2647b030e5
Photo:c82f58bc-833e-49d6-ad06-e931984139d9  <-- How to watch this result?
UserProfile:0c5902f1-6482-4662-99a3-3bca1479e1ef
```
